src/m365/aad/commands/user/user-add.ts

```diff
@@ -0,0 +1,272 @@
+          return `'${args.options.managerUserId}' is not a valid GUID.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['managerUserId', 'managerUserName'],
+        runsWhen: (args) => {
+          return args.options.managerId || args.options.managerUserName;
+        }
+      },
+      {
+        options: ['forceChangePasswordNextSignIn', 'forceChangePasswordNextSignInWithMfa'],
+        runsWhen: (args) => {
+          return args.options.forceChangePasswordNextSignIn || args.options.forceChangePasswordNextSignInWithMfa;
+        }
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.boolean.push('accountEnabled');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Adding user to AAD with displayName ${args.options.displayName} and userPrincipalName ${args.options.userName}`);
+    }
+
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/users`,
+        headers: {
+          'accept': 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          accountEnabled: args.options.accountEnabled || true,
+          displayName: args.options.displayName,
+          userPrincipalName: args.options.userName,
+          mailNickName: args.options.mailNickname ?? args.options.userName.split('@')[0],
+          passwordProfile: {
+            forceChangePasswordNextSignIn: args.options.forceChangePasswordNextSignIn,
+            forceChangePasswordNextSignInWithMfa: args.options.forceChangePasswordNextSignInWithMfa,
+            password: args.options.password ?? this.generatePassword()
+          },
+          givenName: args.options.firstName,
+          surName: args.options.lastName,
+          usageLocation: args.options.usageLocation,
+          officeLocation: args.options.officeLocation,
+          jobTitle: args.options.jobTitle,
+          companyName: args.options.companyName,
+          department: args.options.department,
+          preferredLanguage: args.options.preferredLanguage
+        }
+      };
+
+      const user = await request.post<ExtendedUser>(requestOptions);
+      user.password = requestOptions.data.passwordProfile.password;
+
+      if (args.options.managerUserId || args.options.managerUserName) {
+        const managerRequestOptions: CliRequestOptions = {
+          url: `${this.resource}/v1.0/users/${user.id}/manager/$ref`,
+          headers: {},
+          data: {
+            '@odata.id': `${this.resource}/v1.0/users/${args.options.managerUserId || args.options.managerUserName}`
+          }
+        };
+        await request.put(managerRequestOptions);
+      }
+
+      logger.log(user);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private generatePassword(): string {
```

Comment:
When generating a password, I got the following message: `The specified password does not comply with password complexity requirements. Please provide a different password.` This was because the password only consisted of lowercase and capital letters. No digits or special characters.

Reply:
I've added some logic for this.  Obviously, companies can choose their own password policies. So if they, let's say, require atleast 12 characters, it's impossible for us to know this. This might be an enhancement for the future.